src/IO/ReadBuffer.h

```diff
@@ -115,7 +115,7 @@ class ReadBuffer : public BufferBase
 
     void ignore(size_t n)
     {
-        while (n != 0 && !eof())
+        while (n != 0 && !eof() && pos < working_buffer.end())
```

Comment:
This modification looks strange, most likely unneeded.
Are the modifications in MaterializedMySQL enough?

Reply:
Technically this is not needed, https://github.com/ClickHouse/ClickHouse/blob/master/src/IO/ReadBuffer.h#L77 this assert should notify when impl returns nothing. Interesting why we don't see this assert...
Also found some failed tests locally, converted to draft to test.